repo/manifest/serialized.go

```diff
@@ -11,14 +11,38 @@ import (
 
 type manifest struct {
 	Entries []*manifestEntry `json:"entries"`
+	Version int              `json:"version,omitempty"`
 }
 
 type manifestEntry struct {
 	ID      ID                `json:"id"`
 	Labels  map[string]string `json:"labels"`
 	ModTime time.Time         `json:"modified"`
 	Deleted bool              `json:"deleted,omitempty"`
-	Content json.RawMessage   `json:"data"`
+	Content json.RawMessage   `json:"data,omitempty"`
+	// For index versions 1 and above, this is the content ID the manifest data
+	// resides in.
+	ContentID string `json:"contentID,omitempty"`
```

Comment:
this will add 16 bytes to the structure, can we reuse 'Content' for this?

IIUC old-style Content will always be JSON object `{"field":"value", ....}`

Can we distinguish them with:

```
if (Content[0] == '{') {
  // treat as inline content
} else {
  // treat as ContentID
}
```

Reply:
that's an excellent idea! I only thought about the serialized version of the data. It should be fairly trivial to check the format version and then switch the behavior of the Content field based on that. I think then we could avoid having to check the first character since deserialization of this field is deferred until a manifest is requested, at which point we have a format version associated with it

If we're looking to minimize memory usage, then re-arranging how we track manifests in-memory may also be useful. Right now we're wrapping the existing `manifestEntry` with a struct that associates a format version (4-8 bytes for an int) with every `manifestEntry`. We could instead keep the in-memory representation more like the serialized version and only store the format version once for each set of manifests

Reply:
alright, I played around with trying to use a single version of the version specifier like `committedEntries = map[int]map[ID]*manifestEntry` and it complicated some of the logic quite a bit to expand functions to handle that. For now, I think I'll leave that out of this PR and we can revisit if we feel it's worthwhile